src/Contractors.jl

```diff
@@ -58,85 +58,178 @@ or when max_iter iterations are reached (with an error)
 """
 root(f, x; ϵ, max_iter) = root(f, derivative(f), x; ϵ, max_iter)
 
-function root(f, f′, x; ϵ, max_iter)
+function root(f::Function, f′::Function, x::Interval; ϵ, max_iter)
 	for i in 1:max_iter
+		x_left, x_right = bisect(x)
 		
+		if !(0 ∈ f(x_left))
```

Comment:
Is this order really the one we want? This will default to bisection for most functions, and this should be much slower than Newton (~50 steps rather than ~5).